### PR TITLE
fix azure_rm.py not showing nic info for vmss

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

fix issue https://github.com/ansible/ansible/issues/51944, in vmss nic output, there's no type, so nic model was filtered out. fix it.

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by yungezz** [comment]
ready_for_review

File: lib/ansible/plugins/inventory/azure_rm.py [hunk 1]
```diff
@@ -580,10 +579,11 @@ def __init__(self, nic_model, inventory_client, is_primary=False):
 
         self.public_ips = {}
 
-        for ipc in nic_model['properties']['ipConfigurations']:
-            pip = ipc['properties'].get('publicIPAddress')
-            if pip:
-                self._inventory_client._enqueue_get(url=pip['id'], api_version=self._inventory_client._network_api_version, handler=self._on_pip_response)
+        if nic_model.get('properties') and nic_model.get('properties').get('ipConfigurations'):
```
Comment (zikalino) [inline lib/ansible/plugins/inventory/azure_rm.py]: maybe should be get('properties', {}).get('ipConfigurations')
so it will be shorter

**Review by zikalino — APPROVED** [review]

**Comment by yungezz** [comment]
customer verified the PR in issue  #51944

**Comment by Fred-sun** [comment]
@nitzmahone Please help review and push this PR move forward when you're available! Thanks!

**Comment by Fred-sun** [comment]
+azure label

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**